### PR TITLE
Template and testing updates for new RR design

### PR DESCRIPTION
**Pull request by neiljerram**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by neiljerram** [comment]
@caseydavenport There are pinning conflicts and updates that will be needed before this can be merged, but that's all mechanical stuff, and I'd appreciate your review of the substantive changes here.

**Comment by neiljerram** [comment]
Thanks @caseydavenport for your review.  As noted in the confd PR, I think the key factor for your suggestion is whether we want to enhance the calicoctl UI now, and if so how - so please let's talk about that.

**Comment by caseydavenport** [comment]
@neiljerram I think, as you suggested, we should keep as-is for now. We can make incremental improvements to the calicoctl UI, etc as follow on work if so desired. I'd rather not hold this up on that!

File: Makefile [hunk 1]
```diff
@@ -175,7 +173,7 @@ LIBCALICO_VERSION?=$(shell git ls-remote git@github.com:projectcalico/libcalico-
 FELIX_REPO?=github.com/projectcalico/felix
 FELIX_VERSION?=$(shell git ls-remote git@github.com:projectcalico/felix master 2>/dev/null | cut -f 1)
 CONFD_REPO?=github.com/projectcalico/confd
-CONFD_VERSION?=$(shell git ls-remote git@github.com:projectcalico/confd master 2>/dev/null | cut -f 1)
+CONFD_VERSION?=$(shell git ls-remote git@github.com:projectcalico/confd feature-internal-rr 2>/dev/null | cut -f 1)
```
Comment (caseydavenport) [inline Makefile]: Need to update this to master once merged.

Comment (caseydavenport) [reply Makefile]: I meant to say once the confd PR was merged, which is has now, so we can update this :)

**Pull request merged**